src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,360 @@
+        certificate_hash = getCertificateFingerprint(sas_th_config['serverCert'])
+        self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                   'url': sas_th_config['url']})
+
+        response = self.TriggerFullActivityDumpAndWaitUntilComplete(sas_th_config['serverCert'], sas_th_config['serverKey'])
+        # check dump message format
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        # an array for each record type
+        cbsd_dump_data = []
+        ppa_dump_data = []
+        esc_sensor_dump_data = []
+        
+        # step 8 and check   
+        # download dump files and fill corresponding arrays
+        for dump_file in response['files']:
+            self.assertContainsRequiredFields("ActivityDumpFile.schema.json",
+                                               dump_file[0])
+            downloaded_file = None
+            if dump_file['recordType'] != 'CoordinationEvent':                
+                downloaded_file = self._sas.DownloadFile(dump_file['url'],\
+				    sas_th_config['serverCert'], sas_th_config['serverKey'])
+                datetime.strptime(downloaded_file['startTime'],\
+                                                     '%Y-%m-%dT%H:%M:%SZ')
+                datetime.strptime(downloaded_file['endTime'],\
+                                                     '%Y-%m-%dT%H:%M:%SZ')
+            if dump_file['recordType'] ==  'cbsd':
+                cbsd_dump_data.append(downloaded_file['recordData'])   
+            elif dump_file['recordType'] ==  'esc_sensor':
+                esc_sensor_dump_data.append(downloaded_file['recordData'])
+            elif dump_file['recordType'] ==  'zone':
+                ppa_dump_data.append(downloaded_file['recordData'])
+            else:
+                self.assertEqual('CoordinationEvent', dump_file['recordType'])
+        
+        # verify the length of records equal to the inserted ones
+        self.assertEqual(len(config['registrationRequests']), len(cbsd_dump_data))
+        self.assertEqual(len(config['ppas']), len(ppa_dump_data))
+        self.assertEqual(len(config['escSensors']), len(esc_sensor_dump_data))
+        # verify the schema of record and first two parts of PPA record Id  
+        for ppa_record in ppa_dump_data:
+            self.assertContainsRequiredFields("zoneData.schema.json", ppa_record)              
+            self.assertEqual(ppa_record['id'].split("/")[0], 'zone')
+            self.assertEqual(ppa_record['id'].split("/")[1], self._sas._sas_admin_id)
+            del ppa_record['id']
+        # verify that the injected ppas exist in the dump files
+		# TODO: check that the PPAs overlap nearly entirely, rather than requiring exactly the same vertices.
+        for index, ppa in enumerate(config['ppas']):
+            del ppa['id']
+            exist_in_dump = False
+            for ppa_record in ppa_dump_data:			           
```

Comment:
I think you can just do:

```
exist_in_dump = False
for ppa_record in ppa_dump_data:
  exist_in_dump = exist_in_dump or ppa_record == ppa
self.assertTrue(exist_in_dump)
```

Reply:
I think we need to sort the data before comparing them, otherwise the SAS UUT will fail the test if it put the elements of any array parameter of the records in the dump with different order(palId or features or points of GeoJson), right?
I tried it with the unit test in  helper_unit_testcase and the case of 
compareDict(ppa, same_ppa_with_different_order) doesn't work with normal comparison(==)

Reply:
Hmm, this is a very interesting point. I think some of the lists (e.g. PAL IDs, cluster list) can come in any order while other lists must be in the original order (GeoJSON is the only one that comes to mind).

As a note, I think this only applies to lists since:
```
In [4]: a = {1:2, 3:[5,6]}

In [5]: b = {3:[5,6], 1:2}

In [6]: c = {1:2, 3:[6,5]}

In [7]: a == b
Out[7]: True

In [8]: a == c
Out[8]: False
```

So I agree that we need something more nuanced than `==` but I think we still need to adjust the implementation a bit.

Other than the GeoJSON, is there any list where order matters? If not, perhaps we can check the zone, remove it from the object, then use `compareDicts()`? I'm not really sure what the cleanest solution is... maybe it is to just check the fields directly.

Reply:
for the points of the ppa contour I think they don't need to have exactly the same order, but both lists should contain CCW ordered points, but if they don't begin with the same point they will not be the same list, right?

i think the both following lists of points are accepted for a ppa contour:
A: [-120, 40], [-120,39][-119,39][-119,40]
B:  [-119,39][-119,40],[-120, 40],[-120,39]

right?

Reply:
I agree that they don't need to have exactly the same order, they will not be permutations either: the last point of the list always needs to be equal to the first point for the GeoJSON to be valid, so if the SAS circular-shifts the points internally, it will actually end up with a different set after converting to GeoJSON: [a, b, c, a] vs. [b, c, a, b].

Since we expect to switch to polygon comparison ASAP, I suggest the following approach:
 - Carefully think about whether there are any lists that *must* be in the same order (I can't think of any offhand).
 - If there are none, continue using your compareDict() function but rename it so that the subtlety of sorting the lists is more obvious. Maybe something like "compareDictWithUnorderedLists()".
 - Either:
    (a) Do an order-specific comparison on the points within the GeoJSON, then remove the zone from the dict before comparing.
    (b) Continue checking the zone inside compareDict() but add a comment that this is wrong (but also that it will be fixed ASAP by the polygon comparison code).
 - Add a note into the docstring of the testcase that there is a known bug which prevents this from matching the WF spec exactly OR create an issue on GitHub to track the problem (I personally find it very easy to lose track of these things so I don't want to risk forgetting).

Probably (b) is the best approach as it is the least effort and we expect the fix to come before we do much testing of SASes anyway. What do you think?